tests/rados/test_inconsistency_osd_epoch.py

```diff
@@ -0,0 +1,139 @@
+            **pool_target_configs,
+        )
+        pool_name = pool_target_configs["pool_name"]
+        log.info(f"Created the pool {pool_name} to generate inconsistent objects")
+
+        # Creating omaps
+        if not pool_obj.fill_omap_entries(pool_name=pool_name, **omap_target_configs):
+            log.error(f"Omap entries not generated on pool {pool_name}")
+            return 1
+        log.debug("Completed writing omap entries onto the pool")
+
+        # Fetching all objects and selecting one object to generate inconsistency
+        obj_list = rados_obj.get_object_list(pool_name)
+        log.debug(f"All the objects added onto the pool are : {obj_list}")
+
+        oname = random.choice(obj_list)
+        log.debug(
+            f"Selected object {oname} at random to generate inconsistent objects."
+        )
+
+        primary_osd = rados_obj.get_osd_map(pool=pool_name, obj=oname)["acting_primary"]
+        log.debug(f"The object stored in the primary osd number-{primary_osd}")
+
+        # getting the OSD epoch before generating inconsistent objects
+        cmd = "ceph osd dump"
+        init_epoch = rados_obj.run_ceph_command(cmd=cmd)["epoch"]
+        log.info(f"Initial epoch before generating inconsistency is {init_epoch}")
+
+        # Create inconsistency objects
+        pg_id = rados_obj.create_inconsistent_object(pool_name, oname)
+        log.debug(f"PG ID {pg_id} has inconsistent object generated on it.")
+
+        # Checking for inconsistency in the PG list
+        inconsistent_pg_list = rados_obj.get_inconsistent_pg_list(pool_name)
+        if any(pg_id in search for search in inconsistent_pg_list):
+            log.info(f"Inconsistent PG is{pg_id}  ")
+        else:
+            log.error("Inconsistent PG is not generated")
+            raise Exception("Inconsistent object not generated error")
+
+        cmd = "ceph osd dump"
+        epoch_incon = rados_obj.run_ceph_command(cmd=cmd)["epoch"]
+        log.info(f"OSD epoch when Inconsistent object is generated : {epoch_incon}")
+
+        log.debug(
+            f"Marking the primary OSD with inconsistency, OSD.{primary_osd} down and out"
+        )
+        rados_obj.change_osd_state(action="stop", target=primary_osd)
+        rados_obj.run_ceph_command(cmd=f"ceph osd out {primary_osd}")
+
+        # Checking for the inconsistent pg's
+        inconsistent_pg_list = rados_obj.get_inconsistent_pg_list(pool_name)
+        if any(pg_id in search for search in inconsistent_pg_list):
+            log.error(
+                "Inconsistent pg exists in the cluster even after the OSD epoch was changed with OSD down"
+            )
+            raise Exception("Inconsistent PG on cluster Error")
+        log.info("Inconsistent pg does not exist on the cluster post epoch change.")
+
+        cmd = "ceph osd dump"
+        epoch_osd = rados_obj.run_ceph_command(cmd=cmd)["epoch"]
+        log.info(f"OSD epoch when Inconsistent object is generated : {epoch_osd}")
```

Comment:
Can the logger be improved to show difference in line 81, line 100 and line 122 w.r.t OSD at the start, OSD being out and back In to make debugging easier.

Reply:
Done. Updated log lines.